packages/dumb/src/components/trends/cbg/cbg-slice/cbg-slice-animated.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2017-2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import cx from 'classnames'
+import React, { FunctionComponent } from 'react'
+import { spring, TransitionMotion } from '@serprex/react-motion'
+
+import { springConfig } from '../../../../models/constants/animation.constants'
+
+import styles from './cbg-slice-animated.css'
+import { CbgSliceTransitionMotionInterpolate } from '../../../../models/animation.model'
+import CbgSliceSegment from './cbg-slice-segment'
+import { TrendsDisplayFlags } from '../../../../models/trends-display-flags.model'
+import { getRangeSegments } from './cbg-slice-animated.util'
+
+interface CbgSliceAnimatedProps {
+  datum: {
+    firstQuartile?: number
+    id: string
+    max?: number
+    median?: number
+    min?: number
+    msX: number
+    ninetiethQuantile?: number
+    tenthQuantile?: number
+    thirdQuartile?: number
+  }
+  displayFlags: TrendsDisplayFlags
+  showingCbgDateTraces: boolean
+  sliceWidth: number
+  tooltipLeftThreshold: number
+  topMargin: number
+  xScale: Function
```

Comment:
Depending on who merges their PR first, you could reuse the type `XScale` here 😄 I don't know the signature of `yScale`, if it's the same we could adapt the name of the type